torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -533,6 +534,12 @@ XLAGraphExecutor::SyncTensorCollection XLAGraphExecutor::CollectSyncTensors(
   // The force_ltc_data controls aliasing compilation, so effectively the same
   // graph with on/off force_ltc_data should not match, hash wise.
   coll.hash = torch::lazy::MHash(config.force_ltc_data);
+  // Ensure the compilation environment and git revision are reflected in the
+  // hash.
+  coll.hash = torch::lazy::HashCombine(
+      coll.hash, runtime::GetComputationClient()->HashCompilationEnv());
+  coll.hash =
+      torch::lazy::HashCombine(coll.hash, torch::lazy::StringHash(XLA_GITREV));
```

Comment:
cc @yeounoh, I wonder if we need to include whether or not we're executing in SPMD mode here. I found that the hash is different for SPMD vs non-SPMD, but I'm not sure if that will be the case generally.

Reply:
Let's add it. We don't really want SPMD and nonSPMD hash to collide .

Reply:
Done

Reply:
FWIW It should already be dependent to SPMD via sharding hash.

Reply:
https://github.com/pytorch/xla/pull/5803/files#r1414428856